src/lua/serializer.h

```diff
@@ -243,6 +252,15 @@ struct luaL_field {
 	bool compact;                /* a flag used by YAML serializer */
 };
 
+/**
+ * Get the next field from the table in sorted order.
+ * @param L Lua stack.
+ * @param cfg Serializer configuration.
+ * @retval 0 if there are no more fields in the table.
+ */
+int
+luaL_next_field(struct lua_State *L, struct luaL_serializer *cfg);
```

Comment:
Nit: [lua_next](https://pgl.yoyo.org/luai/i/lua_next)'s description explicitly tells what is popped from the stack and what is pushed to it. I find it useful and I would do it here as well.

Reply:
I've added a detailed comment for the `next_field` function:
```c
/**
 * @brief Get the next field from the table in sorted order.
 *
 * Pops a key and a 'key index' from the stack, and pushes a key, 'key-index',
 * value for the next element in a table provided as a third argument on stack.
 *
 * If cfg->encode_key_order is set, this function iterates over the table in the
 * oreder defined by this array. During this process the 'key-index' is the
 * index of the current key in cfg->encode_key_order array. If there are no more
 * elements in the cfg->encode_key_order array this function iterates over the
 * unvisited elements of the table in an arbitrary order.
 *
 * During the iterating this function may push a copy (which may be modified by
 * the iterator) of the provided table on the stack which will be popped when
 * there are no more elements left in the table.
 *
 * When the iterator reaches the last element of the table this function returns
 * 0 and doesn't push anything on the stack, leaving the original table.
 *
 * A typical traversal looks like this:
 *
 * // cfg->encode_key_order is an array with the order of the keys
 * // table is on the top of the stack
 * lua_pushnil(L);  // first key index
 * lua_pushnil(L);  // first key
 * while (luaL_next_field(l, cfg) != 0) {
 *     printf("%s - %s\n",
 *            lua_typename(L, lua_type(L, -2)),  // key
 *            lua_typename(L, lua_type(L, -1))); // value
 *     lua_pop(L, 1); // remove value, keep key and key index for next iteration
 * }
 *
 * @param L Lua stack.
 * @param cfg Serializer configuration.
 * @retval 0 if there are no more fields in the table.
 */
int
luaL_next_field(struct lua_State *L, struct luaL_serializer *cfg);
```

Reply:
> oreder

Thanks for the update!